lib/Installation/ConfirmInstallation.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Handles Confirm Installation Page.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+package Installation::ConfirmInstallation;
+use Installation::Popups::AcceptPopup;
+use Installation::Popups::OKPopup;
+use strict;
+use warnings;
+use testapi qw(save_screenshot record_info);
+use YuiRestClient;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my $self = shift;
```

Comment:
this is very hard to maintain, you are doing POMs joining two popups in one page, in POM you get some part of the screen or all the screen, but not two screens that sometimes one happen and sometimes happens both...whoever needs to change this code would have a lot of doubts. It would be wiser to do the condition at the level of test layer (in other words, when I recommend to not use conditional logic in test layer, I mean that I even recommend less to use condition logic in the framework, it is like shooting yourself in the foot).

Confirm installation is just one thing, you are dealing with something that should be called `ConfirmAllBeforeInstallation`, which is quite blurry for this corner case, which is a popup that only happens in our automation in the way that we wrongly test the update, it is nto even a functional thingy, notice you need those screenshot in the middle here because of that, you can do the screenshot at the level of test.

In summary I would expect to see this conditional logic well visible in `tests/installation/confirm_installation.pm`

Reply:
I am still using this kind of logic, only because the accept button, like the ok button, can be seen in other Pages etc. I could easily include the accept button in the LicencePopup, instead of having it in a separate Popup, but in case we find it elsewhere, we have to repeat the code. If you think it's uncommon and not worth the trouble, I'll implement it in LicencePopup